spec/features/votes_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+RSpec.describe "Votes" do
+  let!(:user) { create :user, email: 'bob@test.com', username: 'Bob' }
+  let!(:article) { create :article }
+  let!(:article_with_comment) { create :article }
+  let!(:comment) { create :comment, commentable: article_with_comment }
+
+  before do
+    visit '/session/new'
+    fill_in 'Email', with: 'bob@test.com'
+    fill_in 'Password', with: 'password'
+    click_button 'Log in'    
+  end
+
+  describe "upvoting an article" do
+    it "adds article score" do
+      visit "/articles/#{article.id}"
+      within("#article_#{article.id}") do
+        click_link "Upvote"
+      end
+      within("#article_#{article.id}") do
+        expect(page).to have_content("Upvoted")
+        expect(page).to have_content("Score: 1")
+      end
+    end
+  end
+
+  describe "downvoting an article" do
+    it "subtracts article score" do
+      visit "/articles/#{article.id}"
+      within("#article_#{article.id}") do
+        click_link "Downvote"
+      end
+      within("#article_#{article.id}") do
+        expect(page).to have_content("Downvoted")
+        expect(page).to have_content("Score: -1")
+      end
+    end
+  end
+
+  describe "upvoting a comment" do
+    it "adds comment score" do
+      visit "/articles/#{article_with_comment.id}"
+      within("#comment_#{comment.id}") do
+        click_link 'Upvote'
+      end
+      within("#comment_#{comment.id}") do
+        expect(page).to have_content("Upvoted")
+        expect(page).to have_content("Karma: 1")
+      end
+    end
```

Comment:
Not a big deal, mostly a style thing, but I like to separate all the initial clicking around stuff into a `before` block and then have the `it` only contain the `expect` statements:


```ruby
describe "upvoting a comment" do
  before do
    visit "/articles/#{article_with_comment.id}"

    within("#comment_#{comment.id}") do
      click_link 'Upvote'
    end
  end

  it "adds comment score" do
    within("#comment_#{comment.id}") do
      expect(page).to have_content("Upvoted")
      expect(page).to have_content("Karma: 1")
    end
  end
end
```

I like this because it clearly separates the test preparation from the actual test. It also makes it less weird to have to do `within` twice, because they're in separate places.